tribe-ext-relabeler.php

```diff
@@ -0,0 +1,322 @@
+ */
+class Main extends Tribe__Extension {
+
+	/**
+	 * Caches labels that are retrieved from the database.
+	 *
+	 * @var array {
+	 *      @type $option_name string Full text for the altered label
+	 * }
+	 */
+	protected $label_cache = [];
+
+	/**
+	 * Setup the Extension's properties.
+	 *
+	 * This always executes even if the required plugins are not present.
+	 */
+	public function construct() {
+		$this->add_required_plugin( 'Tribe__Events__Main' );
+		$this->set_url( 'https://theeventscalendar.com/extensions/extensions/change-labels-events-venues-organizers/' );
+	}
+
+	/**
+	 * Get this plugin's options prefix.
+	 *
+	 * Settings_Helper will append a trailing underscore before each option.
+	 *
+	 * @see \Tribe\Extensions\Example\Settings::set_options_prefix()
+	 *
+	 * @return string
+	 */
+	private function get_options_prefix() {
+		return (string) str_replace( '-', '_', PLUGIN_TEXT_DOMAIN );
+	}
+
+	/**
+	 * Get Settings instance.
+	 *
+	 * @return Settings
+	 */
+	private function get_settings() {
+		if ( empty( $this->settings ) ) {
+			$this->settings = new Settings( $this->get_options_prefix() );
+		}
+
+		return $this->settings;
+	}
+
+	/**
+	 * Extension initialization and hooks.
+	 */
+	public function init() {
+
+		load_plugin_textdomain( 'PLUGIN_TEXT_DOMAIN', false, basename( dirname( __FILE__ ) ) . '/languages/' );
+
+		if ( ! $this->php_version_check() ) {
+			return;
+		}
+
+		$this->class_loader();
+
+		$this->get_settings();
+
+		// Events.
+		add_filter( 'tribe_event_label_singular', [ $this, 'get_event_single' ] );
+		add_filter( 'tribe_event_label_singular_lowercase', [ $this, 'get_event_single_lowercase' ] );
+		add_filter( 'tribe_event_label_plural', [ $this, 'get_event_plural' ] );
+		add_filter( 'tribe_event_label_plural_lowercase', [ $this, 'get_event_plural_lowercase' ] );
+
+		// Venues.
+		add_filter( 'tribe_venue_label_singular', [ $this, 'get_venue_single' ] );
+		add_filter( 'tribe_venue_label_singular_lowercase', [ $this, 'get_venue_single_lowercase' ] );
+		add_filter( 'tribe_venue_label_plural', [ $this, 'get_venue_plural' ] );
+		add_filter( 'tribe_venue_label_plural_lowercase', [ $this, 'get_venue_plural_lowercase' ] );
+
+		// Organizers.
+		add_filter( 'tribe_organizer_label_singular', [ $this, 'get_organizer_single' ] );
+		add_filter( 'tribe_organizer_label_singular_lowercase', [ $this, 'get_organizer_single_lowercase' ] );
+		add_filter( 'tribe_organizer_label_plural', [ $this, 'get_organizer_plural' ] );
+		add_filter( 'tribe_organizer_label_plural_lowercase', [ $this, 'get_organizer_plural_lowercase' ] );
+	}
+
+	/**
+	 * Check if we have a sufficient version of PHP. Admin notice if we don't and user should see it.
+	 *
+	 * @link https://theeventscalendar.com/knowledgebase/php-version-requirement-changes/ All extensions require PHP 5.6+.
+	 *
+	 * @return bool
+	 */
+	private function php_version_check() {
+		$php_required_version = '5.6';
```

Comment:
I think this one can be defined on this class a constant to make it more clear.

```
const public REQUIRED_PHP_VERSION = '5.6'
```

Reply:
I left this unchanged for now, as we don't have it in the template.